website/content/docs/concepts/stateful-deployments.mdx

```diff
@@ -53,6 +53,21 @@ If the scheduler cannot find a node that has the right volume ID present
 stopped and the job will remain in `pending` status until it's either stopped
 by the user or a feasible node is found.
 
+## Task Group Host Volume Claims
+
+Nomad stores associations of host volume IDs and their task groups (as well as
+job IDs, volume names and namespaces) in its state store. There is an API
+available to [list] and [delete] these associations. If a task group host volume
+claim is deleted, the task group will get a different volume ID (assuming the
+old volume ID is unavailable and another feasible host volume with matching name
+is), but once placed, Nomad will once again record the claim in its state store.
+This is to say, that claim deletions should only ever be used in operational
+emergency situations, or while debugging problems with nodes (for example in
+situations where operators want to drain a node with a stateful deployment on
+it, but do not want to stop the job).
```

Comment:
The example given -- node drain -- is for many clusters a totally normal operation to replace old clients with new ones, so this being "emergency" or "debugging" only makes sense if the reader is fully aware of the intended use. My read of this page falls a little short of that.

I think it could be clearer earlier on this page that being sticky to a host volume means that it is inherently sticky to the _host_ that volume is _permanently_ tied to, because it's not sticky to a volume _name_ (which could be on many nodes), it's sticky to the _id_ (tied to a single, specific node, forever).

That information is all present here, but the _host_ part of group->volume->_host_ is sorta between the lines, and I think we could make it more explicit.

Reply:
does this https://github.com/hashicorp/nomad/pull/25288/commits/04abe38d05b66e686a513eaa303eebfd1e020b4d help, Daniel?